File: tests/algorithms/x/PHP/conversions/excel_title_to_column.php
<?php
ini_set('memory_limit', '-1');
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
$__start_mem = memory_get_usage();
$__start = _now();
  $letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
  function excel_title_to_column($title) {
  global $letters;
  $result = 0;
  $i = 0;
  while ($i < strlen($title)) {
  $ch = substr($title, $i, $i + 1 - $i);
  $value = 0;
  $idx = 0;
  $found = false;
  while ($idx < strlen($letters)) {
  if (substr($letters, $idx, $idx + 1 - $idx) == $ch) {
  $value = $idx + 1;
  $found = true;
  break;
}
  $idx = $idx + 1;
};
  if (!$found) {
  $panic('title must contain only uppercase A-Z');
}
  $result = $result * 26 + $value;
  $i = $i + 1;
};
  return $result;
};
  function main() {
  global $letters;
  echo rtrim(json_encode(excel_title_to_column('A'), 1344)), PHP_EOL;
  echo rtrim(json_encode(excel_title_to_column('B'), 1344)), PHP_EOL;
  echo rtrim(json_encode(excel_title_to_column('AB'), 1344)), PHP_EOL;
  echo rtrim(json_encode(excel_title_to_column('Z'), 1344)), PHP_EOL;
};
  main();
$__end = _now();
$__end_mem = memory_get_peak_usage();
$__duration = max(1, intdiv($__end - $__start, 1000));
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;

File: tests/algorithms/transpiler/FS/conversions/excel_title_to_column.fs
// Generated 2025-08-07 10:31 +0700

exception Break
exception Continue

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let letters: string = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
let rec excel_title_to_column (title: string) =
    let mutable __ret : int = Unchecked.defaultof<int>
    let mutable title = title
    try
        let mutable result: int = 0
        let mutable i: int = 0
        try
            while i < (String.length (title)) do
                try
                    let ch: string = title.Substring(i, (i + 1) - i)
                    let mutable value: int = 0
                    let mutable idx: int = 0
                    let mutable found: bool = false
                    try
                        while idx < (String.length (letters)) do
                            try
                                if (letters.Substring(idx, (idx + 1) - idx)) = ch then
                                    value <- idx + 1
                                    found <- true
                                    raise Break
                                idx <- idx + 1
                            with
                            | Continue -> ()
                            | Break -> raise Break
                    with
                    | Break -> ()
                    | Continue -> ()
                    if not found then
                        failwith ("title must contain only uppercase A-Z")
                    result <- (result * 26) + value
                    i <- i + 1
                with
                | Continue -> ()
                | Break -> raise Break
        with
        | Break -> ()
        | Continue -> ()
        __ret <- result
        raise Return
        __ret
    with
        | Return -> __ret
and main () =
    let mutable __ret : unit = Unchecked.defaultof<unit>
    try
        let __bench_start = _now()
        let __mem_start = System.GC.GetTotalMemory(true)
        printfn "%d" (excel_title_to_column ("A"))
        printfn "%d" (excel_title_to_column ("B"))
        printfn "%d" (excel_title_to_column ("AB"))
        printfn "%d" (excel_title_to_column ("Z"))
        let __bench_end = _now()
        let __mem_end = System.GC.GetTotalMemory(true)
        printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)

        __ret
    with
        | Return -> __ret
main()
